hipe4ml/TrainingAndTesting/generalized_analysis.py

```diff
@@ -0,0 +1,131 @@
+'''Module containing the class to perform the training and the testing of the models '''
+
+
+import pandas as pd
+import uproot
+from sklearn.model_selection import train_test_split
+
+
+class GeneralizedAnalysis:
+    ''' Class for training and testing machine learning models starting from ROOT TTrees filled with the candidates
+    '''
+
+    def __init__(
+            self, mc_file_name, data_file_name, sig_selection=0, bkg_selection=0,
+            cent_class=((0, 10),
+                        (10, 30),
+                        (30, 50),
+                        (50, 90))):
+
+        self.cent_class = cent_class.copy()
+        self.n_events = [0, 0, 0, 0]
+
+        self.df_signal = uproot.open(mc_file_name)['SignalTable'].pandas.df()
+        self.df_generated = uproot.open(mc_file_name)['GenTable'].pandas.df()
```

Comment:
If possible I would avoid to have the dataframe for the generated particles as mandatory, since there could be cases in which it is not needed